operator/controllers/seldondeployment_controller.go

```diff
@@ -52,18 +52,21 @@ import (
 const (
 	ENV_DEFAULT_ENGINE_SERVER_PORT      = "ENGINE_SERVER_PORT"
 	ENV_DEFAULT_ENGINE_SERVER_GRPC_PORT = "ENGINE_SERVER_GRPC_PORT"
+	ENV_CONTROLLER_ID                   = "CONTROLLER_ID"
 
 	DEFAULT_ENGINE_CONTAINER_PORT = 8000
 	DEFAULT_ENGINE_GRPC_PORT      = 5001
 
 	AMBASSADOR_ANNOTATION = "getambassador.io/config"
+	LABEL_CONTROLLER_ID   = "controller-id"
```

Comment:
I would suggest adding some prefix to "controller-id" for label key, such as "seldon.io/controller-id" to distinguish it from possible label naming collision.